### PR TITLE
Compute class

### PR DESCRIPTION
**Pull request by Daniel-Devory**
Created class definition for compute board interface

### CODE REVIEW

File: shepherd_bms/include/compute.h [hunk 1]
```diff
@@ -1,6 +1,23 @@
 #ifndef COMPUTE_H
 #define COMPUTE_H
 
-#include "datastructs.h"
```
Comment (nwdepatie) [inline shepherd_bms/include/compute.h:4]: We probably want to keep this include in there, it's going to contain the data structures that we utilize to store data

File: shepherd_bms/include/compute.h [hunk 2]
```diff
@@ -1,6 +1,23 @@
...
+{
+    private:
+        int fanSpeed;
+
+    public: 
+        ComputeInterface();
+
+        ~ComputeInterface();
+
+        void enableCharging();
```
Comment (nwdepatie) [inline shepherd_bms/include/compute.h:16]: enableCharging should get a boolean passed as a parameter called something like "isEnabled" or something. You might want to make a private boolean that we write this state to for now, and when we implement charging logic it'll become a bit more complex

File: shepherd_bms/include/compute.h [hunk 3]
```diff
@@ -1,6 +1,23 @@
...
+
+        void setFanSpeed(int newFanSpeed);
+
+        int getPackCurrent();
```
Comment (nwdepatie) [inline shepherd_bms/include/compute.h:20]: return type of int is okay for now, but we might want to change it to a uint8_t later on to save space

**Review by nwdepatie — CHANGES_REQUESTED** [review]

**Review by nwdepatie — APPROVED** [review]
I cleaned up some of the code and I think it is good to merge in now, but note in the future we should start naming the branches starting with the ticket number

File: shepherd_bms/include/compute.h [hunk 4]
```diff
@@ -2,5 +2,59 @@
...
+
+class ComputeInterface
+{
+    private:
+        uint8_t fanSpeed;
+
+        /**
+         * @todo These might need to be changed depending on the charging ticket
+         */
+        const uint8_t startChargingMsg[];
```
Comment (mccauleyma) [inline shepherd_bms/include/compute.h:17]: Yeah this is almost certainly not how it works. There is a constant message that is sent with the current/voltage that we want delivered

Comment (mccauleyma) [reply shepherd_bms/include/compute.h:17]: If this is a placeholder anyways though we can leave it for now. But may already want to rename to "chargingCmdMsg" or something

File: shepherd_bms/include/compute.h [hunk 5]
```diff
@@ -2,5 +2,59 @@
 #define COMPUTE_H
 
 #include "datastructs.h"
+#include "nerduino.h"
 
+using namespace std;
+
+class ComputeInterface
+{
+    private:
```
Comment (mccauleyma) [inline shepherd_bms/include/compute.h:11]: This is probably just a lack of my knowledge on the architecture, but are we declaring BMS-wide faults and data here eventually, or is there another higher level that will define that? Stuff like saving the current reading, controlling DCL, etc

**Review by mccauleyma — APPROVED** [review]
Nothing blocking, but curious for answers to my questions

File: shepherd_bms/include/compute.h (hunk 5 above)
Comment (nwdepatie) [reply shepherd_bms/include/compute.h:11]: This class is just meant as an interface for the IO of the compute board, we shouldn't be storing any internal states in it really

File: shepherd_bms/include/compute.h (hunk 4 above)
Comment (nwdepatie) [reply shepherd_bms/include/compute.h:17]: Yeah I might change it when I get back and look into what the charger message actually contains

**Pull request merged**